Backend

PersianBlue opened the pull request:
Created a Game Clock to keep track of the game's time
Ran npm install to fix some gatsby dependency issues

PersianBlue committed:
Changed name of index.js to Button.js in components

PersianBlue committed:
Created GameClock, updated gatsby dependencies

Natreve commented:
Components use their folder names to reference them, best practice would be to put it back to what it was originally. This allows you to access to component by like so `import Button from "components/button"` instead of `import Button from "components/button/Button"` It is also important for individuality  styling components with their own style sheet module.

You will notice before I had a button folder with an accompanied css.module.scss file

PersianBlue commented:
> 

Ahh that explains it, and makes more sense, many thanks

PersianBlue committed:
Added reset button| migrated controller

Added a functional reset button that stops the timer when clicked (so that the timer doesn't just restart from 0), but this required migrating the stop timer feature from ClockController to GameClock so ClockController was removed and the functions transferred with their corresponding buttons and variables

PersianBlue committed:
Added Team component

Created a Team component that displays team name, score, and provides two buttons to increment and decrement score

Natreve commented:
The components you should have

1. Button (Variation on button style should be controlled based on property value passed into the property "type" such as circular, rounded, square, etc)
2. Clock (A component that is a clock, should be able to control this clock based on a reference)

Natreve commented on src/components/team/Team.js:
```diff
@@ -0,0 +1,26 @@
+import React from 'react'
+import {useState} from 'react'
+import Button from "../button/Button"
```
This should read `import Button from "../button"` because the button component would have had the index.js and the css.module.scss you do not need to name the javascript file Button

PersianBlue replied:
Oh so leave the js files in each component folder as index.js then?

Natreve commented on src/pages/index.js:
```diff
@@ -1,13 +1,16 @@
 import React from "react";
 import Button from "../components/button/Button";
 import GameClock from "../components/gameclock/GameClock"
+import Team from "../components/team/Team"
```
This should read `import Team from "../team" `because the team component would have had the index.js and the css.module.scss you do not need to name the javascript file Team

PersianBlue committed:
Add: throwtimer, change seconds

created throwtimer component which displays countdown timer upon clicking
after the countdown ends, whistle sound is played
Added change seconds button, that brings up a prompt to accept input
Also added input validation to check if no valid input is entered

Natreve commented:
Almost there, for organisation I would add the images and audio files in an assets folder. So you would have a folder structure like so 
assets/images
assets/audio

PersianBlue commented:
> 

Makes sense, got it

PersianBlue committed:
Squashed commit of the following:

commit 27ca1d95323b35697b2dc35e48ac9990d8bba7f5
Author: Natreve <natreve.ag20@gmail.com>
Date:   Wed Apr 20 09:38:49 2022 -0500

    fixed screenshot

PersianBlue committed:
Added gradient background, with noise

Added svg file for grainy texture/noise
Set background in global css for body types to a linear gradient (white to grey as in photo), then used that background for the <main> tag in index.js for the main page
